src/UI/Art/SpriteArtist/UnitSpriteSet.java

```diff
@@ -3,119 +3,95 @@
-   * @param width The width of each frame in the sprite.
-   * @param height The height of each frame in the sprite.
+   * Fetch a sprite sheet for each animation state and colorize it
+   * @param fileFinder Mapping from action type to file name
+   * @param coColors What colors the end sprites should use
    */
-  public UnitSpriteSet(BufferedImage spriteSheet, int width, int height, ColorPalette coColors)
+  public UnitSpriteSet(Function<AnimState, String> fileFinder, ColorPalette coColors)
   {
-    int h = 0;
-    int action = ACTION_IDLE; // This variable relies on the ordering of the ACTION variables defined above.
-
     try
     {
-      if( null != spriteSheet && spriteSheet.getWidth() >= width && spriteSheet.getHeight() >= height )
+      for( int action = 0; action < AnimState.values().length; ++action )
       {
-        while (height <= spriteSheet.getHeight() && action <= ACTION_DIE)
+        String filestr = fileFinder.apply(AnimState.values()[action]);
+        BufferedImage spriteSheet = SpriteLibrary.loadSpriteSheetFile(filestr);
+        if( null != spriteSheet )
         {
-          Sprite spr = new Sprite(spriteSheet.getSubimage(0, h, spriteSheet.getWidth(), height), width, height);
+          // Assume sub-sprites are squares that fill out the height of the source image
+          Sprite spr = new Sprite(spriteSheet, spriteSheet.getHeight(), spriteSheet.getHeight());
           sprites[action] = spr;
-          ++action;
-        }
-        if( action < ACTION_DIE )
-        { // We didn't load all animations we wanted - our sprite set is incomplete. Default the rest to IDLE.
-          sprites[action] = sprites[ACTION_IDLE];
-          ++action;
         }
       }
-      else
-      { // No sprite sheet provided? Just make stuff up.
-        System.out.println("WARNING! Continuing with placeholder images.");
-        sprites[ACTION_IDLE] = new Sprite(null, width, height);
-        for( int i = ACTION_IDLE + 1; i <= ACTION_DIE; ++i )
-        {
-          sprites[i] = sprites[ACTION_IDLE];
-        }
-        turnDone = sprites[ACTION_IDLE];
-      }
     }
-    catch (RasterFormatException rfe)
+    catch (Exception e)
     {
-      // Something went wrong. Just make something up and hope nobody notices.
-      // Use the IDLE action if it exists, otherwise we are going with a black rectangle.
-      Sprite defaultSprite = (action > ACTION_IDLE) ? sprites[ACTION_IDLE] : new Sprite(null, width, height);
-
-      System.out.println("WARNING: RasterFormatException in UnitSpriteSet constructor.");
+      System.out.println("WARNING: Exception hit in UnitSpriteSet constructor:" + e);
       System.out.println("WARNING:   Attempting to continue.");
-      for( ; action <= ACTION_DIE; ++action )
-      {
+    }
+
+    // Handle the case of having one sideways move direction and not the other defined
+    if( null == sprites[AnimState.MOVEEAST.ordinal()] && null != sprites[AnimState.MOVEWEST.ordinal()] )
```

Comment:
Where did you get the move sprites? I'm gonna infer that they all game in west-facing format, and you just decided to future-proof that variable.

Reply:
A spritesheet. And yeah.